File: 프로그래머스/LV1/두개뽑아서더하기/Solution.cpp
//문제 정보
// https://programmers.co.kr/learn/courses/30/lessons/68644?language=cpp

// 내부에서 중복을 알아서 걸러주는 Set이라는 자료구조를 사용했다면 
// 좀더 쉽게 풀었을 것이라 생각된다. 공부하자. 

#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer;
    
    int arr[201] ={0}; 
    
    for(int i= 0 ; i< numbers.size(); i++){
        for(int j= i+1 ; j< numbers.size(); j++){
            arr[numbers[i]+numbers[j]] = 1; 
        }
    }
    
    for(int i=0 ; i< 201; i++){
        if(arr[i] !=0){
            answer.push_back(i); 
        }
    }
    return answer;
}

File: 프로그래머스/LV1/두개뽑아서더하기/Solution.java
import java.util.*; 
import java.util.ArrayList; 
import java.util.Collections;



//문제 정보
// https://programmers.co.kr/learn/courses/30/lessons/68644

class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = {};
        Set<Integer> set = new HashSet<Integer>();
        for(int i = 0 ; i< numbers.length ; i++){
            for(int j= i+1; j< numbers.length; j++){
                set.add(numbers[i]+numbers[j]); 
            }
        }
        
        List<Integer> list = new ArrayList<Integer>(set); 
        
        Collections.sort(list); 
        
        answer = new int[list.size()];
        
        int size=0;
        for(int temp : list){

          answer[size++] = temp;

        }
        
        return answer;
    }
}
